jsx/src/components/ServerDashboard/ServerDashboard.test.js

```diff
@@ -9,39 +10,44 @@ import {
   getByText,
   getAllByRole,
 } from "@testing-library/react";
-import { HashRouter, Switch } from "react-router-dom";
+import { HashRouter, Routes, Route, useSearchParams } from "react-router-dom";
+// import { CompatRouter,  } from "react-router-dom-v5-compat";
 import { Provider, useSelector } from "react-redux";
 import { createStore } from "redux";
 // eslint-disable-next-line
 import regeneratorRuntime from "regenerator-runtime";
 
 import ServerDashboard from "./ServerDashboard";
 import { initialState, reducers } from "../../Store";
-import * as sinon from "sinon";
-
-let clock;
 
 jest.mock("react-redux", () => ({
   ...jest.requireActual("react-redux"),
   useSelector: jest.fn(),
 }));
+jest.mock("react-router-dom", () => ({
+  ...jest.requireActual("react-router-dom"),
+  useSearchParams: jest.fn(),
+}));
 
-var serverDashboardJsx = (spy) => (
-  <Provider store={createStore(mockReducers, mockAppState())}>
-    <HashRouter>
-      <Switch>
-        <ServerDashboard
-          updateUsers={spy}
-          shutdownHub={spy}
-          startServer={spy}
-          stopServer={spy}
-          startAll={spy}
-          stopAll={spy}
-        />
-      </Switch>
-    </HashRouter>
-  </Provider>
-);
+const serverDashboardJsx = (props) => {
```

Comment:
This file has a lot of changes, mostly around making this function reusable, rather than needing the whole `<Provider...` repeated many times.